<launch>
  <arg name="ARM_TYPE" default="$(env ARM_TYPE)" />
  <node pkg="nodelet" type="nodelet" name="standalone_nodelet"  args="manager" output="screen"/>

  <node pkg="nodelet" type="nodelet" name="ArmManager" args="load kinova_control_manager/ArmManager standalone_nodelet" output="screen">
    <rosparam file="$(find kinova_control_manager)/config/$(arg ARM_TYPE).yaml" />

    <!-- output commands to arm driver -->
    <remap from="~home_out" to="$(arg ARM_TYPE)_driver/in/home_arm" />
    <remap from="~end_effector_pose_out" to="$(arg ARM_TYPE)_driver/pose_action/tool_pose/goal" />

      <!-- output to the mrs gripper -->
    <remap from="~grip_out" to="kinova/gripper/grip" />
    <remap from="~ungrip_out" to="kinova/gripper/ungrip" />
    
    <!-- status input from arm driver and gripper -->
    <remap from="~joint_angles_in" to="$(arg ARM_TYPE)_driver/out/joint_angles" />
    <remap from="~joint_torques_in" to="$(arg ARM_TYPE)_driver/out/joint_torques" />
    <remap from="~end_effector_pose_in" to="$(arg ARM_TYPE)_driver/out/tool_pose" />
    <remap from="~gripper_diagnostics_in" to="kinova/gripper/gripper_diagnostics" />

    <!-- manager status -->
    <remap from="~arm_status_out" to="kinova/arm_manager/status" />

    <!-- provided commands -->
    <remap from="~home_in" to="kinova/arm_manager/home_arm" />
    <remap from="~soft_home_in" to="kinova/arm_manager/soft_home_arm" />
    <remap from="~prepare_gripping_in" to="kinova/arm_manager/prepare_gripping" />
    <!-- <remap from="~align_arm_in" to="kinova/arm_manager/align_arm" /> -->
    <remap from="~aim_at_in" to="kinova/arm_manager/aim_at_target" />
    <remap from="~goto_in" to="kinova/arm_manager/goto" />
    <remap from="~goto_relative_in" to="kinova/arm_manager/goto_relative" />

  </node>
</launch>
